aptos-move/framework/aptos-token/sources/token_event_utils.move

```diff
@@ -0,0 +1,318 @@
+/// This module provide utils to create and emit token events
+module aptos_token::token_event_utils {
+    use std::string::String;
+    use std::signer;
+    use aptos_framework::event::{Self, EventHandle};
+    use aptos_framework::account;
+    use std::option::Option;
+    use aptos_std::any::Any;
+    use std::option;
+
+    friend aptos_token::token;
+
+    //
+    // Collection mutation events
+    //
+
+    /// Event emitted when collection description is mutated
+    struct CollectionDescriptionMutateEvent has drop, store {
+        creator_addr: address,
+        collection_name: String,
+        /// new description
```

Comment:
nit: Might as well just name this field new_description

Reply:
Same for other fields below

Reply:
After 2nd thought, I cannot think of a use case where they need the old values. For example, marketplace want to use the new_uri to update image or show new royalty of token, etc.

Do you have a use-case in mind of storing the old value?

Reply:
It's a good practice to store the old value in general as this can:
1. Allow keeping track of changes history. Users can technically do some historical query of data before an event, but that adds a lot more complexity to off chain systems. It's much easier if the events keep track of the before and after
2. Many systems want to show the changes (diffs). I can imagine this being the case for an NFT as well where users want to see what changed (from which value to which value).
3. It's impossible to add new fields to an existing event in the future, and this is generally the case for smart contracts on all blockchains. Therefore, the best practice is to add as many potentially useful fields as possible, unless it significantly increases gas cost.

Reply:
added old values.